### DISCUSSION TITLE
Dual Lens Camera in Frigate?

### DISCUSSION THREAD

**Discussion by dz4news** [blakeblackshear/frigate#6955] [Ask A Question] [answered]
Has anyone tried a camera like this in Frigate? Will it work?

I have a side of my house that a 180 degree field of view would work great... I dont mind if it splits into two images, as long as I can see both of them in Frigate.

EmpireTech 4K Full-Color 2×4MP Dual-Lens 1/1.8" CMOS Smart Splicing Network IP Camera Outputs 180° Wide Angle, Support Two-Way Talk IPC-Color4K-T180 (White)

**Comment by NickM-27** [comment, marked as answer]
I use https://www.bhphotovideo.com/c/product/1705511-REG/hikvision_colorvu_ds_2cd2387g2p_lsu_sl_8mp_network.html/ and it works well in frigate, appears as a single camera. Will work even better in 0.13 with the new birdseye layout.

I would not recommend the empiretech panoramic because its default resolution squishes the wide angle meaning objects do not appear normally, you can look at https://youtu.be/twnZayxPjzY?t=361 to see what I mean

**Reply by wout1976** [reply]
I get a message no frames have been received. Can u share your config and settings in camera?

**Reply by NickM-27** [reply]
it would be easier for you to share your config and logs

**Reply by wout1976** [reply]
voor-180:
    ffmpeg:
      hwaccel_args: preset-vaapi
      output_args:
        record: -f segment -segment_time 20 -segment_format mp4 -reset_timestamps
          1 -strftime 1 -c:v copy -c:a aac
      inputs:
      - path: rtsp://admin:GGGGGG@192.168.178.134:554/cam/realmonitor?channel=2&subtype=2
        roles:
        - detect
        - rtmp
      - path: rtsp://admin:GGGGGG@192.168.178.134:554/cam/realmonitor?channel=2&subtype=2
        roles:
        - record
    detect:
      width: 1920
      height: 536
      fps: 20
    objects:
      track:
      - person
      - dog
      filters:
        person:
          threshold: 0.7
    ui:
      order: 1
    snapshots:
      enabled: true
      retain:
        default: 8
        objects:
          person: 8
    record:
      enabled: true
      retain:
        days: 8
      events:
        retain:
          default: 8

**Reply by wout1976** [reply]
And Gortc
 Voor-180:
    - rtsp://admin:GGGGGGG@192.168.178.134:554/cam/realmonitor?channel=2&subtype=2

**Reply by NickM-27** [reply]
Those are the wrong URLs for hikvision cameras. Should be rtsp://{FRIGATE_RTSP_USER}:{FRIGATE_FRONT_PW}@192.168.50.152/streaming/channels/101 for main stream and rtsp://{FRIGATE_RTSP_USER}:{FRIGATE_FRONT_PW}@192.168.50.152/streaming/channels/102 for sub stream

**Reply by wout1976** [reply]
Still no image, but how are your camera settings in Hikvision: mine 5120X1440  15 fps H264

Frigate
inputs:
      - path: rtsp://admin:ggggggggq0k!@192.168.178.134/streaming/channels/101
        roles:
        - detect
        - rtmp
      - path: rtsp://admin:GGGGG@192.168.178.134/streaming/channels/102
        roles:
        - record
    detect:
      width: 1920
      height: 536
      fps: 20

**Reply by NickM-27** [reply]
you have the paths backwards, 102 is the sub stream. 

and yes I have mine set as h264 and 20 fps

**Reply by NickM-27** [reply]
you should include logs if you have issues

**Reply by wout1976** [reply]
Yesss! It works by changing the streams. Thanks!

**Comment by dz4news** [comment]
I have the Hikvision on order.. any particular settings in the config for frigate to work with it?

**Reply by NickM-27** [reply]
I guess it depends how you want to use it. I use the main stream for detection with a detect resolution of `2560x720` but if you do that you have to use the cameras main stream with h265 because the resolution is too large to be used with h264. I also just got the night settings tuned well but that is of course dependent on where it is installed.

**Reply by imdeepjee** [reply]
> I have the Hikvision on order.. any particular settings in the config for frigate to work with it?

Hi, can you please give an update on Hikvision camera? Does it work well in Frigate?

**Reply by dz4news** [reply]
I ended up returning the camera and never tried it in frigate.
